spring-cloud-stream-binder-kinesis/src/main/java/org/springframework/cloud/stream/binder/kinesis/provisioning/KinesisStreamProvisioner.java

```diff
@@ -140,7 +141,8 @@ public ConsumerDestination provisionConsumerDestination(String name, String grou
 			}
 			catch (ResourceNotFoundException ex) {
 				if (!this.configurationProperties.isAutoCreateStream()) {
-					throw ex;
+					throw new ProvisioningException("The stream [" + stream
+							+ "] was not found. Auto creation is disabled (autoCreateStream=false)",ex);
```

Comment:
Well, let's rephrase it like this:
`"The stream [" + stream + "] was not found and auto creation is disabled.`

And also, please, a while space before `ex`